[add] skeleton outline

ting-dev-coder opened the pull request:
create some necessary skeleton outline titles

ting-dev-coder committed:
[add] skeleton outline

ash-teach commented:
Can you comment on the purpose of the `Pull Requests` section? I'm hesitant to add what is likely to become a quickly outdated section in official documentation, especially given that PRs are tracked through GitHub (i.e. you can always visit the [PR tab](https://github.com/nic-dgl104-winter-2024/pattern-library/pulls) and see all current and historical PRs).

ash-teach commented:
Adding a license is a great idea. However, I think this would be better handled through a separated LICENSE.md (as is typical with most well-managed open source projects). Identifying the right license might be an interesting challenge. I think adding this as a new issue is a good idea. See https://opensource.org/licenses as a reference.

ting-dev-coder commented:
Yes , I can see all current and historical PRs, is this part of purpose of the Pull Requests section?

ash-teach commented on README.md:
```diff
@@ -1,2 +1,14 @@
+
+
+
+## Pull Requests 
```
I don't understand the purpose of this section - can you explain? My assumption is that you intend to include a list of PRs here? But we can track this through the standard GitHub tools (i.e. the [PR tab](https://github.com/nic-dgl104-winter-2024/pattern-library/pull)).

Did you have something else in mind for this section?

ting-dev-coder replied:
For the Pull Requests section, I think we can encourage people to create pull requests and maybe some instruction for contributors to follow?

ting-dev-coder replied:
For example , the content for this section could be 
" We welcome pull requests! Please follow the [Pull Request Template](.github/PULL_REQUEST_TEMPLATE.md) "

ash-teach replied:
I see. So a section to encourage contributions. **Let's call it `Contributions` instead,** since that's a bit more inclusive of what contributions might look like.

If this section grows we can consider breaking it out into a separate `CONTRIBUTING.md` document.

ting-dev-coder committed:
add and revise skeleton outline

[modify] replace the Pull Requests section with Contributions
[add] articles section
[remove] License section

ting-dev-coder commented on README.md:
For the Articles / Websites section, we could link some useful websites or articles about design pattern 
for example : [DESIGNPATTERNS](https://refactoring.guru/design-patterns/catalog)

ash-teach replied:
Yes, I think that is reasonable. In fact, as an academic effort, we should include a more strict from of this as a set of official references. We'll leave it as-is for now, but we may want to explore changing this heading down the road.
